datadog/resource_datadog_dashboard.go

```diff
@@ -1212,7 +1274,12 @@ func buildTerraformAlertGraphDefinition(datadogDefinition datadogV1.AlertGraphWi
 		terraformDefinition["title_align"] = *v
 	}
 	if v, ok := datadogDefinition.GetTimeOk(); ok {
-		terraformDefinition["time"] = buildTerraformWidgetTime(*v)
+		// Set to deprecated field if that's what is used in the config, otherwise, set in the new field
+		if _, ok := k.GetOkWith("time"); ok {
+			terraformDefinition["time"] = buildTerraformWidgetTime(*v)
```

Comment:
nit: for consistency and clarity, should we replace the function name by `buildTerraformWidgetTimeDeprecated`